CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+
+In the development repository, you will find 3 types of branches :
+- The main branch
+- Team's branch
+- Features ( or bugfix ) branches
+
+You must code in a feature branch. This feature branch will eventually end up in a PR to your team's branch. At this level, only your team's members are required to review. This PR is annotated with **PR 1** in the image above.
+
+After accepting multiples PRs by your team's members, your team's branch will get to a point where you feel it's ready to close an issue ( more on that later ). At this point, you must open a PR from you team's branch to the main branch ( **PR 2** ). This PR must be reviewed by at least 5 reviewers before getting marked as accepted.
+
+When all the teams are satisfied with the work in the main branch, and everybody agrees on the correctness of what's been done, a new PR, from this fork to the parent repo ( the main repo ) ( **PR 3** ), will be opened to be discussed with the parent's repo maintainer. 
+
+## How to commit
+
+First, let's list the basic rules :
+- A commit message has a title ( summary ), body and footer
+- The title should be 50 characters maximum
+- Capitalise the first letter of title, don't put a period ( . ) at the end
+- Put a blank line between the title, body and footer
+- Wrap the commit body at 72 characters
+- Try to write the commit body as **bullets** ( optional )
+- Use imperative in title ( **Add** instead of **Added**, for example )
+- Describe what and why, but not how ( the purpose, not the implementation )
+- Footer contains issue number, if applicable ( #78 )
+
+> For more information about the 50/72 characters rule, please visit [this link.](https://dev.to/noelworden/improving-your-commit-message-with-the-50-72-rule-3g79#:~:text=The%2050%2F72%20Rule%20is,pulled%20out%20of%20a%20hat.)
+
+Your commits should follow the [conventional commits specification.](https://www.conventionalcommits.org/en/v1.0.0/#summary)
+
+Please be mindful of this conventionn when writing your commits. CI jobs will check if your commits follow this convention, since these commits might later be used to create changelog using parsers that understand this convention specifically.
+
+## When to make an Issue, and what's its goal ?
+An issue should be opened when new functional needs are defined, a new bug is discovered, or a documentation resource is going to be created. You **must** open an issue before opening the corresponding branch.  An issue should not be technical, and should describe what the end user will get. You open an issue when either one of the following condition is true :
+- You want to describe what your team is currently working on : this issue is meant to be merge from your team's branch to the main branch
+- You want to describe features that you are currently working on : this issue is meant to be merge from a feature branch to your team's branch
+
```

Comment:
Again, a lot of this is specific to your class workflow, not so much to external contributors.

And the rule for deciding about "When to make an Issue" should be simple: A PR should address an issue. There must be at least one commit in your PR that closes/fixes an issue. No exception.